lib/colorado_lottery.rb

```diff
@@ -0,0 +1,39 @@
+class ColoradoLottery
+  attr_reader :registered_contestants,
+              :winners,
+              :current_contestants
+  def initialize(registered_contestants = {},
+                winners = [],
+                current_contestants = {})
+    @registered_contestants = registered_contestants
+    @winners = winners
+    @current_contestants = current_contestants
+  end
+
+  def interested_and_18?(contestant, game)
+    interested = contestant.game_interests.include?(game.name)
+    contestant.age >= 18 && interested
+  end
+
+  def can_register?(contestant, game)
+    return false unless interested_and_18?(contestant,game)
+    !contestant.out_of_state? || game.national_drawing?
+  end
+
+  def register_contestant(contestant, game)
+    if can_register?(contestant, game)
+      if registered_contestants[game.name]
+        @registered_contestants[game.name] << contestant
+      else
+        contestants = []
+        @registered_contestants[game.name] = contestants << contestant
```

Comment:
setting contestants and shoveling all in one line is not ideal in terms of style--you'd never really see this. i'd do this something like this: 


```
  def register_contestant(contestant, game)
    if registered_contestants[game.name] == nil
      registered_contestants[game.name] = []
    end
    registered_contestants[game.name] << contestant
  end
```